lib/parser.js

```diff
@@ -381,20 +381,32 @@ const args = [
           'session unless overridden by received capabilities.'
   }],
 
+  [['--allow-insecure'], {
+    dest: 'allowInsecure',
+    defaultValue: [],
+    type: parseAllowInsecureFeatures,
+    required: false,
+    example: 'execute-webdriver,adb-shell',
```

Comment:
This value is the subject of discussion. I would put names of the particular mobile commands here.Also, the maximum length of command line parameters is limited. that is why it also makes sense to enumerate possible values in a file rather than as cmdline argument. The third problem I would like to mention is blacklisting vs whitelisting. I mean one could still provide `relaxedSecurityEnabled`, but disable particular commands with `denyInsecure` arg

Reply:
So @mykola-mokhnach to make sure I'm understanding you, you are saying:

1. We should keep `--relaxed-security` as a "turn on all insecure features" flag
2. We should have `--allow-insecure` as a whitelist (so can never be used with `--relaxed-security`
3. We should have `--deny-insecure` as a blacklist (so must be used with `--relaxed-security`)
4. We should optionally read feature names from a file and not just CLI. What is the right format for this kind of file, do you think?

I'm happy to implement this system if it's what we want. In this case I think it'd be a good idea to make a function in BaseDriver like `isFeatureAllowed(featureName)`, which does all the checking based on these three flags, so that individual drivers only ever need to call `isFeatureAllowed`, to keep their code clean and future-proof. Sound good?

Reply:
i went ahead and implemented that method in the most recent version of https://github.com/appium/appium-base-driver/pull/328

Reply:
> We should have --allow-insecure as a whitelist (so can never be used with --relaxed-security

almost. These args should have priorities, where `relaxed-security` has the lowest one and `deny-insecure` has the highest one. So, for example, if one says `--relaxed-security --allow-insecure shell --deny-insecure shell` then no features are going to be enabled, because:

- `allow-insecure` overrides `relaxed-security` and only allows `shell`
- `deny-insecure` overrides everything else and denies `shell` completely

Reply:
> make a function in BaseDriver like isFeatureAllowed(featureName)

Reply:
> What is the right format for this kind of file, do you think

I would say each feature in the new line should be ok. Empty or unsupported features can be simply ignored